common/app/services/newsletters/EmailEmbedLifecycle.scala

```diff
@@ -0,0 +1,36 @@
+package services.newsletters
+
+import app.LifecycleComponent
+import common.JobScheduler
+import play.api.inject.ApplicationLifecycle
+
+import java.util.concurrent.Executors
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutorService, Future}
+
+class EmailEmbedLifecycle(appLifecycle: ApplicationLifecycle, jobs: JobScheduler, emailEmbedAgent: EmailEmbedAgent)
+    extends LifecycleComponent {
+
+  implicit val newsletterClientExecutionContext: ExecutionContextExecutorService =
+    ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor())
```

Comment:
Has this been copied from other similar components of this app?

Otherwise I think it's okay to get the execution context from Play directly, and receive it during the construction (wiring) of this class

Reply:
I took it from here https://github.com/guardian/frontend/blob/ce59d97ef3e6b4a6ee6d8b215032fda904d26333/onward/app/feed/OnwardJourneyLifecycle.scala#L24 but that's the only other example of this that I can find. Maybe it's better to take it from the wiring instead?